orc8r/cloud/go/services/streamer/providers/providers.go

```diff
@@ -33,34 +38,90 @@ func (p *MconfigProvider) GetStreamName() string {
 }
 
 func (p *MconfigProvider) GetUpdates(gatewayId string, extraArgs *any.Any) ([]*protos.DataUpdate, error) {
-	resp, err := configurator.GetMconfigFor(gatewayId)
+	res, err := configurator.GetMconfigFor(gatewayId)
 	if err != nil {
-		return nil, errors.Wrap(err, "error getting mconfig from configurator")
+		return nil, errors.Wrap(err, "get mconfig from configurator")
 	}
 
-	if extraArgs != nil {
-		// Currently, only use of extraArgs is mconfig hash
-		receivedDigest := &protos.GatewayConfigsDigest{}
-		if err := ptypes.UnmarshalAny(extraArgs, receivedDigest); err == nil {
-			glog.V(2).Infof("Received, generated config digests: %v, %v\n",
-				receivedDigest,
-				resp.Configs.Metadata.Digest.Md5HexDigest,
-			)
-			return mconfigToUpdate(resp.Configs, resp.LogicalID, receivedDigest.Md5HexDigest)
-		}
-	}
+	// TODO(8/5/20): revert below once we send proto descriptors from mconfig_builders
 
-	return mconfigToUpdate(resp.Configs, resp.LogicalID, "")
+	//if extraArgs != nil {
+	//	// Currently, only use of extraArgs is mconfig hash
+	//	receivedDigest := &protos.GatewayConfigsDigest{}
+	//	if err := ptypes.UnmarshalAny(extraArgs, receivedDigest); err == nil {
+	//		glog.V(2).Infof(
+	//			"Received, generated config digests: %v, %v\n",
+	//			receivedDigest,
+	//			res.Configs.Metadata.Digest.Md5HexDigest,
+	//		)
+	//		return mconfigToUpdate(res.Configs, res.LogicalID, receivedDigest.Md5HexDigest)
+	//	}
+	//}
+	//
+	//return mconfigToUpdate(res.Configs, res.LogicalID, "")
+
+	return bytesMconfigToUpdate(res.Configs, res.LogicalID)
 }
 
-func mconfigToUpdate(configs *protos.GatewayConfigs, key string, digest string) ([]*protos.DataUpdate, error) {
-	// Early/empty return if gateway already has config that would be sent here
+func mconfigToUpdate(configs *protos.GatewayConfigs, logicalID string, digest string) ([]*protos.DataUpdate, error) {
+	// Early/empty return if gateway already has this config
 	if digest == configs.Metadata.Digest.Md5HexDigest {
 		return []*protos.DataUpdate{}, streamer.EAGAIN // do not close the stream, there were no changes in configs
 	}
 	marshaledConfig, err := protos.MarshalJSON(configs)
+	if err != nil {
+		return nil, errors.Wrap(err, "marshal gateway mconfig")
+	}
+	return []*protos.DataUpdate{{Key: logicalID, Value: marshaledConfig}}, nil
+}
+
+// TODO(8/5/20): revert below once we send proto descriptors from mconfig_builders
+
+const gatewayConfigsTemplate = `
```

Comment:
this is a funky, use `json.RawMessage` instead and create your own struct that marshals the same way as we expect the proto to